content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -175,6 +175,12 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+We have published the
+[JSON Schema](https://github.com/iterative/pipeline-schema) for `dvc.yaml` files
+that helps code editors and IDEs to provide auto-completion and
+validation/linting and to verify proper structure of `dvc.yaml` locally and/or
+in CI.
```

Comment:
@skshetry would be great to put some links to VS Code and Pycharm is possible on how to install this kind of schema as an example.

Reply:
It's explained in the README of https://github.com/iterative/pipeline-schema

Reply:
it's better to include a sentence here - easier to search, easier to read, good for SEO, etc.

Reply:
Maybe in https://dvc.org/doc/install/plugins and link there?

Reply:
it can be a separate entry in the plugins indeed.

Reply:
@shcheklein @jorgeorpinel, I just wanted to mention it briefly to see if there were any interest and get suggestions from the community before I submit to the catalog of https://github.com/SchemaStore/schemastore.
The installation guide is enough for now, I think. 

> it can be a separate entry in the plugins indeed.

But, this is a good point. We'll anyway need to provide a link to the yaml extension, so I could briefly mention it there (but, probably without mentioning "how-to-install").

Reply:
>  I just wanted to mention it briefly to see if there were any interest and get suggestions from the community before I submit to the catalog of SchemaStore/schemastore

I would say we should submit this - no need to wait. Relying on this small note as a hypothesis test would be not enough - it's hidden too much.

Reply:
@shcheklein, sure, I'll try to submit this week. Till then, I'll kept this as a draft.

Reply:
Added to the schemastore. You should be able to just open dvc.yaml and get the schema on Pycharm and VSCode (with YAML extension). For some reason, I'm getting `Connection timed out` on Pycharm, would be great if you guys could confirm that it works without any effort. @shcheklein @jorgeorpinel

Reply:
Hey @skshetry I see your contribution there (https://github.com/SchemaStore/schemastore/pull/1187/files). Nice, thanks! But I'm not sure how to add or use that on the IDE. On my VSCode, dvc.yaml files just open a YAML files. Same for PyCharm I assume (I don't even have it rn but I've used it before and can download it). Of course I could try and figure it out 😋  but my point is maybe you can write a draft guide in https://dvc.org/doc/install/plugins as mentioned above? I'd be happy to take this over from that point.

Reply:
For VSCode, you just need to install [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) extension and just reload/restart the editor. Then, open `dvc.yaml` file. Based on your network connection, it might take some time but try typing "st" and `Ctrl + Space` to force suggestions.

For pycharm, I'm having some trouble (sometime it works automatically, sometime it just throws me error). For pycharm, 
open a dvc.yaml file. Then, you should get either "No schema available" or "Schema: dvcyaml` written on the bottom status bar. If on "No schema available", click on it and select "dvcyaml", otherwise you are good to go and start typing.

> If you do not see "dvcyaml" in the list, you might need to opt for auto-download schema.

Reply:
> try typing "st" and Ctrl + Space

Nice, this works on VSCode. Didn't even need the RedHAT YAML extension, it's supported out-of-box it seems.

Reply:
Ah never mind you were right, I needed the YAML extension. The previous autocomplete was basic IntelliSense based on existing words in the doc.